src/brod_client.erl

```diff
@@ -270,6 +277,22 @@ handle_info(Info, State) ->
                           [?MODULE, self(), State#state.client_id, Info]),
   {noreply, State}.
 
+handle_call({start_producer, TopicName, ProducerConfig}, _From, State) ->
+  case State#state.producers_sup of
+    undefined ->
```

Comment:
how about remove this clause: `maybe_start_producers_sup([])` and always have the supervisor started (no children if empty list)